packages/form-core/src/FormApi.ts

```diff
@@ -980,6 +980,17 @@ export class FormApi<
       return
     }
 
+    this.store.batch(() => {
+      void (
+        Object.values(this.fieldInfo) as FieldInfo<any, TFormValidator>[]
+      ).forEach((field) => {
+        field.instance?.options.listeners?.onSubmit?.({
+          value: field.instance.state.value,
+          fieldApi: field.instance,
+        })
+      })
+    })
+
```

Comment:
So after a little tinkering I came up with this solution.. please, let me know what you think

Reply:
That looks great! :)

The only question I have is about that `batch`, are we expecting the user to change form state inside listeners?

Reply:
Sounds good, if I get some free time in the coming weeks I'll be sure to come back and take a pass at it.